src/main/java/org/elasticsearch/cluster/metadata/MetaDataUpdateSettingsService.java

```diff
@@ -168,16 +168,9 @@ public void onFailure(Throwable t) {
 
     public void updateSettings(final UpdateSettingsClusterStateUpdateRequest request, final ActionListener<ClusterStateUpdateResponse> listener) {
         ImmutableSettings.Builder updatedSettingsBuilder = ImmutableSettings.settingsBuilder();
-        for (Map.Entry<String, String> entry : request.settings().getAsMap().entrySet()) {
-            if (entry.getKey().equals("index")) {
-                continue;
-            }
-            if (!entry.getKey().startsWith("index.")) {
-                updatedSettingsBuilder.put("index." + entry.getKey(), entry.getValue());
-            } else {
-                updatedSettingsBuilder.put(entry.getKey(), entry.getValue());
-            }
-        }
+        updatedSettingsBuilder.put(request.settings());
+        updatedSettingsBuilder.remove("index");
```

Comment:
It's no longer needed. It was added to handle "index" parameter in the REST API but the rest API is removing this and other parameters for quite a while now. I will remove this line. Thanks!